### PR TITLE
Move benchmark contracts to era_vm

### PR DESCRIPTION
**Pull request by JulianVentura**
This PR moves custom made benchmark contracts sources from zksync-era to era_vm, so its easier to modify them.
It also adds a new benchmark result processing script made in python, which allows us to easily aggregate all benchmark execution times.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -84,6 +84,9 @@ bench-setup: submodules build_bench_contracts $(ZKSYNC_BENCH_TEST_DATA) $(ZKSYNC
 bench:
 	cd $(ZKSYNC_ROOT) && cargo bench --bench criterion "$(lambda|fast_vm|legacy)/(fibonacci|send)"
 
+check-flamegraph:
+	cd $(ZKSYNC_ROOT) && CARGO_PROFILE_BENCH_DEBUG=2 cargo flamegraph --root --bench criterion -- --bench --profile-time 5 "$lambda/fibonacci_rec^"
```
Comment (Oppen) [inline Makefile:88]: I think the `$lambda` is converted to `ambda` for some reason. What does `$l` do here? Is the `$` even necessary?

File: bench_results.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        cleaned_content = re.sub(r'\n(\s+)time:', ' time:', content)
+        for line in cleaned_content.split('\n'):
+            match = re.match(pattern, line)
+            if match:
+                machine, lower_unit, mid_time, mid_unit, upper_unit = match.groups()
+                # Convert mid_time to milliseconds
+                mid_time_ms = convert_to_ms(float(mid_time), mid_unit)
+                machines[machine] = machines.get(machine, 0) + mid_time_ms
+
+    return machines
```
Comment (Oppen) [inline bench_results.py:30]: `criterion` is capable of emitting JSON IIRC. Maybe that could save you parsing.

**Review by Oppen — APPROVED** [review]

**Comment by Oppen** [comment]
Just to make sure, we should check the generated assembly for the contracts to confirm they do what we think they do.

File: Makefile (hunk 1 above)
Comment (JulianVentura) [reply Makefile:88]: No, it is not necessary actually. I had been using it without any problems though, so I think it works ok.

File: bench_results.py (hunk 2 above)
Comment (JulianVentura) [reply bench_results.py:30]: I didn't know that, it might be useful to improve the script. Thanks!

**Comment by JulianVentura** [comment]
LGTM

**Review by fkrause98 — APPROVED** [review]

**Pull request merged**